R/download.R

```diff
@@ -612,6 +615,26 @@ external_download <- function(dataset = NULL, source = NULL, year = NULL,
     if (param$dataset == "mapbiomas_grazing_quality") {
       dat <- readxl::read_excel(temp, sheet = "BD_Qualidade")
     }
+    if(param$dataset == "mapbiomas_mining" & param$geo_level == "country"){
+
+      dat = readxl::read_excel(temp, sheet = "BR")
+    }
+    if(param$dataset == "mapbiomas_mining" & param$geo_level == "state") {
+
+      dat = readxl::read_excel(temp, sheet = "UF")
+    }
+    if(param$dataset == "mapbiomas_mining" & param$geo_level == "biome"){
+
+      dat =  readxl::read_excel(temp, sheet = "BIOME")
+    }
+    if(param$dataset == "mapbiomas_mining" & param$geo_level == "municipality"){
+
+      dat = readxl::read_excel(temp, sheet = "MUN")
+    }
+    if(param$dataset == "mapbiomas_mining" & param$geo_level == "indigenous"){
+
+      dat = readxl::read_excel(temp, sheet = "TI")
+    }
```

Comment:
Pra não ter tantos ifs praticamente repetidos, melhor incluir na função 'external_download' um novo argumento `sheet = NULL`. Você define dentro da própria load_mapbiomas qual o sheet de cada geo_level e então chama `external_download(..., sheet = geo_sheet)`. Uma forma boa de fazer isso é, na load_mapbiomas, montar um vetor

```
sheets <- c(
    "county" = "BR",
    "state" = "UF",
    "biome" = "BIOME",
    etc.
    )
```

fazendo a correspondência entre cada geo_level e a sheet correta. Então é só rodar um `geo_sheet <- dplyr::recode(param$geo_level, !!!sheets)`, que irá transformar o geo_level na sheet correspondente. Aí essa sheet entra de argumento `sheet` da external_download, e esses ifs todos podem virar um if único

```
if (param$dataset == "mapbiomas_mining"){
    dat <- readxl::read_excel(temp, sheet = param$sheet)
}
```

Lembrando que no início da external download esse parâmetro sheet vai ter que ser incluído na lista param() e tudo

Reply:
Outra observação: acho que é mais recomendado usar `readxl::read_xls` ou `readxl::read_xlsx` do que a `read_excel` genérica, que é só um wrapper dessas específicas.